File: src/main/java/org/radarcns/consumer/realtime/condition/ConditionBase.java
package org.radarcns.consumer.realtime.condition;

import java.util.List;
import org.radarcns.config.realtime.ConditionConfig;

public abstract class ConditionBase implements Condition {

  private final List<String> projects;

  public ConditionBase(ConditionConfig config) {
    this.projects = config.getProjects();
  }

  @Override
  public List<String> getProjects() {
    return projects;
  }
}

File: src/main/java/org/radarcns/consumer/realtime/condition/ConditionBase.kt
package org.radarcns.consumer.realtime.condition

import org.radarcns.config.realtime.ConditionConfig

abstract class ConditionBase(
        config: ConditionConfig,
        override val projects: List<String>? = config.projects,
        override val subjects: List<String>? = config.subjects,
        override val subjectIdField: String? = config.projectIdField,
        override val projectIdField: String? = config.subjectIdField,
        override val timeField: String? = config.timeField,
        override val sourceIdField: String? = config.sourceIdField,
) : Condition